pythx/cli/utils.py

```diff
@@ -0,0 +1,247 @@
+
+    :param config_path: The configuration file's path
+    :param staging: A boolean to denote whether to use staging or not
+    :param exit_on_missing: Return if the file is missing
+    :param no_cache: Disable the API result cache
+    :return:
+    """
+    if not path.isfile(config_path):
+        if exit_on_missing:
+            return None
+        # config doesn't exist - assume first use
+        eth_address = environ.get("PYTHX_USERNAME") or click.prompt(
+            "Please enter your Ethereum address",
+            type=click.STRING,
+            default="0x0000000000000000000000000000000000000000",
+        )
+        password = environ.get("PYTHX_PASSWORD") or click.prompt(
+            "Please enter your MythX password",
+            type=click.STRING,
+            hide_input=True,
+            default="trial",
+        )
+        c = Client(
+            eth_address=eth_address,
+            password=password,
+            staging=staging,
+            no_cache=no_cache,
+        )
+        c.login()
+        update_config(config_path=config_path, client=c)
+    else:
+        config = parse_config(config_path, tokens_required=True)
+        c = Client(
+            eth_address=config["username"],
+            password=config["password"],
+            access_token=config["access"],
+            refresh_token=config["refresh"],
+            staging=staging,
+            no_cache=no_cache,
+        )
+    return c
+
+
+def ps_core(config, staging, number):
+    """A helper method to retrieve data from the analysis list endpoint.
+
+    This functionality is used in the :code:`pythx ps`, as well as the :code:`pythx top`
+    subcommands.
+
+    :param config: The configuration file's path
+    :param staging: Boolean to denote whether to use the MythX staging deployment
+    :param number: The number of analyses to retrieve
+    :return: The API response as AnalysisList domain model
+    """
+    c = recover_client(config_path=config, staging=staging)
+    if c.eth_address == "0x0000000000000000000000000000000000000000":
+        click.echo(
+            (
+                "This functionality is only available to registered users. "
+                "Head over to https://mythx.io/ and register a free account to "
+                "list your past analyses. Alternatively, you can look up the "
+                "status of a specific job by calling 'pythx status <uuid>'."
+            )
+        )
+        sys.exit(0)
+    resp = c.analysis_list()
+    # TODO: paginate if too few analyses
+    resp.analyses = resp.analyses[: number + 1]
+    update_config(config_path=config, client=c)
+    return resp
+
+
+def get_source_location_by_offset(filename, offset):
+    """Retrieve the Solidity source file's location based on the source map offset.
+    :param filename: The Solidity file to analyze
+    :param offset: The source map's offset
+    :return: The line number
```

Comment:
In my personal experience having the column number has been really useful in pinpointing where the error is coming from (especially in longer lines)  even if it isn't the most precise.

It might be helpful to still have the column number with a disclaimer. WDYT?

Reply:
I'm not sure how precise the location is. Usually it ends up at a random statement in the line, which could even point the user into a wrong direction. I'd feel more comfortable leaving this out for now until we nail this thing.
From a UX perspective it'd also be better to at some point go to the users and say "Hey, we got this new thing and you can use it now" instead of "Hey, we got this thing, but it's kinda broken and works only sometimes". :laughing:

Reply:
That makes sense. I'll make the column resolution as a part of the source_mapping  issue.